src/bp/core/repositories/events.ts

```diff
@@ -0,0 +1,64 @@
+import * as sdk from 'botpress/sdk'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+
+import Database from '../database'
+import { TYPES } from '../types'
+
+export interface EventRepository {
+  findEvents(fields: Partial<sdk.IO.StoredEvent>, searchParams: sdk.EventSearchParams)
+  deleteBeforeDate(date: Date): Promise<void>
+}
+
+export const DefaultSearchParams: sdk.EventSearchParams = {
+  sortOrder: [{ column: 'createdOn' }],
```

Comment:
It feels a bit odd for the sdk user to know that this filter is a knex orerby query ... a user won't understand why the column keyword exist. (i know you probably let it this way so the user can set asc or desc ) ...

It would be more user friendly to simply pass an array or event just a single string param.

Reply:
I've reused the logic we have in the CMS for elements, all the typing is in the sdk for the params & the search order. It gives much more flexibility. I could also create a second SortOrder and Pick<StoredEvent>, but do we want to do that for each different sort orders?